.github/workflows/validate-workflows.yaml

```
@@ -0,0 +1,49 @@
+name: Validate Github Workflows
+
+on:
+  pull_request: {}
+
+jobs:
+  verify-json-validation:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        workflow:
+          - aks.yaml
+          - bpf-checks.yaml
+          - codeql-analysis.yml
+          - cyclonus-netpol-test.yaml
+          - docs.yaml
+          - eks.yaml
+          - gke.yaml
+          - go-check.yaml
+          - images-legacy-base-releases.yaml
+          - images-legacy-hotfix-releases.yaml
+          - images-legacy-releases.yaml
+          - images-legacy.yaml
+          - images.yaml
+          - kind-1.19.yaml
+          - nightly.yaml
+          - smoke-test-ipv6.yaml
+          - smoke-test.yaml
+          - validate-workflows.yaml
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+      - uses: fabasoad/yaml-json-xml-converter-action@7bc70591754798737773675d47037be8dddefcb4
+        id: yaml2json
+        with:
+          path: '.github/workflows/${{ matrix.workflow }}'
+          from: 'yaml'
+          to: 'json'
+      - uses: DamianReeves/write-file-action@e19fd875ed54f16fc583a3486e62547ce4a5dde8
+        with:
+          path: .github/workflows/${{ matrix.workflow }}.json
+          contents: |
+            ${{ steps.yaml2json.outputs.data }}
+          write-mode: overwrite
+      - name: Validate JSON
+        uses: docker://orrosenblatt/validate-json-action:latest
```

Are we required to use the Docker notation? If not, I would switch back to the regular notation. If yes, I would pin the docker image ;)

---

this image is available only on dockerhub, the author didn't really provide other tags. I will try using sha256 digest for pinning though. https://hub.docker.com/layers/orrosenblatt/validate-json-action/

---

Even if the author did not provide a tag, I think it should still work by directly referencing whatever commit we want from the GH repo:
```suggestion
        uses: OrRosenblatt/validate-json-action@a4773333088b27448e122ccd2992d63383ec2342
```

---

tried it, but it seems like the action was meant to be used only as a docker image. I am going to stay with sha256 digest of the docker image and we can try to look for newer action that is distributed in a more modern fashion.